### PR TITLE
Adding metadata sample

### PR DESCRIPTION
**Pull request by jonparrott**

### CODE REVIEW

File: compute/metadata/main.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            time.sleep(1)
+            continue
+        # [END hanging_get]
+
+        last_etag = r.headers['etag']
+
+        if r.text == 'MIGRATE_ON_HOST_MAINTENANCE':
+            in_maintenance = True
+        else:
+            in_maintenance = False
```
Comment (waprin) [inline compute/metadata/main.py]: in_maintenance =  (r.text == 'MIGRATE_ON_HOST_MAINTENANCE')

Comment (jonparrott) [inline compute/metadata/main.py]: Meh, too clever. This a bit more clear.

File: compute/metadata/main.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+METADATA_HEADERS = {'Metadata-Flavor': 'Google'}
+
+
+def wait_for_maintenance(callback):
+    url = METADATA_URL + 'instance/maintenance-event'
+    last_in_maintenance = False
+    # [START hanging_get]
+    last_etag = 0
+
+    while True:
```
Comment (waprin) [inline compute/metadata/main.py]: probably missing something...looks like this code should terminate when status is false but it doesn't look like logic exists to do so?

Comment (jonparrott) [inline compute/metadata/main.py]: It never terminates. It constantly watches for migration events and calls the callback. The callback can decide to terminate or whatever.

File: compute/metadata/main.py (hunk 1 above)
Comment (waprin) [inline compute/metadata/main.py]: Well I disagree, it's not nested in any way, it's not clever as much as what I consider to be standard Pythonic syntax. BUT /bikeshed, don't want to block code purple work etc so whatever.

**Comment by waprin** [comment]
lgtm

**Pull request merged**